trip and expenses bug in stable version

abhimanyumohan opened the pull request:
HI Dolibarr team,

I found a bug in the trip and expenses . When we click on the list of trips, everything seems fine but on clicking the person name, it shows user details, that are not related to the particular user. Anyway I rectified it.
I solved the issue by adding fk_user to the line 63 in compta->deplacement->list.php
$sql.= " d.rowid, d.type, d.dated as dd, d.km,d.fk_user,";

and change line 138
to
print '<td align="left"><a href="'.DOL_URL_ROOT.'/user/fiche.php?id='.$obj->fk_user.'">'.img_object($langs->trans("ShowUser"),"user").' '.$obj->firstname.' '.$obj->lastname.' '.$obj->name.'</a></td>';

Thanks

eldy commented:
I have 1010 files modified by this PR. 
too much for a simple bug. Does someone can send another PR with just the fix ?

eldy committed:
Fix: Minor fix into intervention card

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix: too much field into list when option FICHINTER_DISABLE_DETAILS is
on.

eldy committed:
Sync french

eldy committed:
Fix: Are description too large

eldy committed:
Fix: Solve conflict of translation key

eldy committed:
Fix: List of status does not match status into class.

eldy committed:
Fix: Bad property

Commit:
Fix: Invoices payments may be older than invoices

eldy committed:
Perf: Avoid duplicate load of project

eldy committed:
Merge remote-tracking branch 'origin/3.6' into develop

Conflicts:
	htdocs/.gitignore
	htdocs/core/lib/functions2.lib.php
	htdocs/install/mysql/migration/repair.sql
	htdocs/langs/en_US/errors.lang

eldy committed:
Doxygen

eldy committed:
Fix: Bad field

eldy committed:
Fix: Avoid explode when separator : is not.

eldy committed:
Merge pull request #1903 from aspangaro/develop-accounting

Patch Accounting Expert module

eldy committed:
Merge pull request #1898 from GPCsolutions/ciforeveryone

Fixed typo

eldy committed:
Merge pull request #1902 from atm-maxime/mko_workonlists

Work on lists

eldy committed:
Merge pull request #1878 from Cubexed/develop

This commit adds a way to filter projects by checking if selected year is in creation/ending range.

eldy committed:
Merge pull request #1896 from FHenry/develop

 Add Dunning into accountancy report

eldy committed:
Merge branch 'develop' of github.com:aspangaro/dolibarr into
aspangaro-develop

Conflicts:
	ChangeLog

eldy committed:
Merge branch 'aspangaro-develop' into develop

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Qual: Uniformize fields name.

eldy committed:
Qual: Uniformize param names. contactidp -> contactid

aspangaro committed:
Uniformize code $_SERVER["PHP_SELF"]

aspangaro committed:
Merge remote-tracking branch 'Upstream/develop' into develop

aspangaro committed:
Qual: Renamed all files & links "liste.php" into "list.php"

aspangaro committed:
Qual: Renamed all files & links "fiche.php" into "card.php"

aspangaro committed:
Delete elements have a link with old module ventilation - Replace today by module Accounting Expert

Commit:
Trad: Syncro es_ES from transifex

Commit:
- Fix: [ bug #1593 ] Spanish Localtax IRPF not being calculated since
3.6.0 in supplier invoices when adding a line

eldy committed:
Qual: Uniformize field names.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge remote-tracking branch 'origin/3.6' into develop

Conflicts:
	htdocs/societe/consumption.php

eldy committed:
Add accountancy language file

eldy committed:
Fix: invalid list of params
Fix: loading of sql menu

eldy committed:
Fix several bugs reported by scrutinizer

Commit:
Minor accountancy updates

Commit:
Trad: Replace ACCOUNTINGEX by ACCOUNTING

eldy committed:
Fix several bugs reported by scrutinizer

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix: sql request with pgsql
Fix: bug reported by scrutinizer
Doxygen
Start to work on multi user assigned to an event.

eldy committed:
Add phpunit for interventions

eldy committed:
Doxygen

aspangaro committed:
Merge remote-tracking branch 'Upstream/develop' into develop-accounting

aspangaro committed:
Merge remote-tracking branch 'Upstream/develop' into develop

FHenry committed:
Better accountancy auto find accountacty code feature (work with several
accountacy plan)

FHenry committed:
Re-Add subtotal line on compta report (accidentaly remove before)

FHenry committed:
Better ressource management on project and task

FHenry committed:
Fix FicheInterTest Call

marcosgdf committed:
Merge branch 'develop' of https://github.com/Dolibarr/dolibarr into oop-pdf

marcosgdf committed:
Refactored generateDocument functions

marcosgdf committed:
Fixed syntax error in commande document generation

marcosgdf committed:
Fixed syntax error problem

eldy committed:
Fix: case into phpunit test case

aspangaro committed:
Renamed all constants COMPTA_* into ACCOUNTING_* and migration of module ACCOUNTINGEX 3.6

aspangaro committed:
compta_mode -> accounting_mode

aspangaro committed:
Changelog

aspangaro committed:
Merge remote-tracking branch 'Upstream/develop' into develop

marcosgdf committed:
Added missing function in Expedition::generateDocument

marcosgdf committed:
Removed 2nd argument of muscadet and canelle model __construct function

marcosgdf committed:
Refactored CommandeFournisseur::generateDocument and FactureFournisseur::generateDocument

marcosgdf committed:
Documented commonObject::commonGenerateDocument

Commit:
ADD extrafields on usergroup

eldy committed:
Missing hook to add button on intervention
Prepare to move location into resource
Can select which king of contact show into contact tab.

eldy committed:
Uniformize export profiles.

eldy committed:
Merge pull request #1913 from atm-alexis/feature_extrafields_on_group

ADD extrafields on usergroup

eldy committed:
Merge branch 'develop' of github.com:aspangaro/dolibarr into
aspangaro-develop

Conflicts:
	htdocs/fichinter/card.php

eldy committed:
Merge branch 'develop' of github.com:aspangaro/dolibarr into
aspangaro-develop

Conflicts:
	htdocs/fichinter/card.php

eldy committed:
Merge branch 'aspangaro-develop' into develop

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into
develop

Conflicts:
	htdocs/user/group/fiche.php

eldy committed:
Merge pull request #1910 from FHenry/develop

Better accountancy auto find accountancy code feature (work with several

eldy committed:
Merge branch 'KreizIT-predefined' into develop

Conflicts:
	ChangeLog

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch 'oop-pdf' of github.com:marcosgdf/dolibarr into
marcosgdf-oop-pdf

Conflicts:
	htdocs/projet/class/project.class.php

eldy committed:
Merge branch 'marcosgdf-oop-pdf' into develop

eldy committed:
Merge pull request #1897 from Cubexed/develop-patch-2

Closed projects are greyed out in project selector

eldy committed:
Trans

eldy committed:
Uniformize field name

eldy committed:
Work on multi user for events.
Fix: Modules must be reloaded after migration.

FHenry committed:
Add feature to order to invoice on supplier part

Commit:
add the clone action for a social contribution
- With option to clone it to the next month by default

eldy committed:
Add field color into table of type of events

eldy committed:
Add a TODO to explain why we keep some extends into class.
Should be removed once php min will be 5.4.

eldy committed:
Fix: Pb into repair function

eldy committed:
Correction repair.sql

eldy committed:
Merge pull request #1917 from atm-alexis/feature_add_clone_to_social_contribution

add the clone action for a social contribution

eldy committed:
Qual: Clean a lot of deprecated code

eldy committed:
Fix: bad phpunit test case

eldy committed:
Doxygen.
Ad field color

FHenry committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

FHenry committed:
set ref_supplier for contract field

Commit:
New: Added hook "formConfirm" and "doActions" for fichinter card

eldy committed:
Fix: duplicate field

rdoursenaud committed:
Fixed invoices webservice

eldy committed:
Merge pull request #1922 from fmarcet/develop

New: Added hook "formConfirm" and "doActions" for fichinter card

eldy committed:
Fix: var not initialized

eldy committed:
Fix: Missing closing tr

tiaris committed:
Adding hook 'suppliercard' on supplier cartd

tiaris committed:
adding hook for addMoreActionsbuttons

FHenry committed:
Fix for travis syntax

hregis committed:
Fix: add generic doc and uniformize code

hregis committed:
Fix: add generic doc and use doc dir instead pdf

hregis committed:
Fix: uniformize code

tiaris committed:
list bof orders to process : status validated or in process, or closed
but not billed

hregis committed:
Fix: add possibility to use element id

hregis committed:
New: add possibility to define a custom error message

tiaris committed:
Add link with order when invoice made for a shioppoing

eldy committed:
Assignation of event to several users is on a very good way for 3.7

eldy committed:
Add FIXME

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix: form must not be into a closing and opening tr tag.

eldy committed:
Merge remote-tracking branch 'origin/3.6' into develop

Conflicts:
	htdocs/projet/activity/list.php

eldy committed:
Merge remote-tracking branch 'origin/3.6' into develop

Conflicts:
	htdocs/admin/stock.php
	htdocs/admin/system/index.php
	htdocs/comm/fiche.php
	htdocs/core/lib/admin.lib.php
	htdocs/core/lib/holiday.lib.php
	htdocs/core/lib/pdf.lib.php
	htdocs/core/login/functions_forceuser.php
	htdocs/core/login/functions_http.php
	htdocs/core/menus/standard/auguria.lib.php
	htdocs/core/modules/livraison/pdf/pdf_typhon.modules.php
	htdocs/filefunc.inc.php
	htdocs/holiday/define_holiday.php
	htdocs/opensurvey/fonctions.php
	htdocs/societe/canvas/actions_card_common.class.php
	test/phpunit/CompanyBankAccountTest.php
	test/phpunit/ExportTest.php
	test/phpunit/FactureTest.php
	test/phpunit/FunctionsLibTest.php
	test/phpunit/PropalTest.php
	test/phpunit/SocieteTest.php
	test/phpunit/WebservicesInvoicesTest.php
	test/phpunit/WebservicesThirdpartyTest.php

eldy committed:
Fix: doxygen
Fix: td balance

eldy committed:
Uniformize code for error reporting after a hook.

hregis committed:
Fix: uniformize code

hregis committed:
Fix: use specific css style instead, this causes a shift problem

eldy committed:
Fix: Missing fields into select

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix: 1 second missing into lastdaytoshow.

eldy committed:
Fix: Link to linked object was not working for invoice and order.
Fix: end date

eldy committed:
Uniformize code.

eldy committed:
Perf enhancement

eldy committed:
Fix: On peruser view, fullday event are reported.
Fix: On peruser view, can start from hour 0.

hregis committed:
New: add doc models management in contract module

FHenry committed:
Add cache excel export on harddrive with
MAIN_USE_FILECACHE_EXPORT_EXCEL_DIR const. It can avoid lot's of stack
overflow on large excel export

FHenry committed:
changelog

hregis committed:
Fix: add model_pdf in fetch of contract object

hregis committed:
Fix: use "doc" instead "pdf" dir and uniformize code

eldy committed:
Fix: Change for php 5.6
New: Management of color into type of events dictionnary (visible into
peruser view).
New: Can disable predefined type of events.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

hregis committed:
Fix: Add current context for avoid method execution in bad context, you
can add this test in your method : eg if($currentcontext != 'formfile')
return;

hregis committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

hregis committed:
Fix: Try to filter by context, use the test with $currentcontext in
method for avoid to run twice (I have not found any other solution!)

eldy committed:
Merge pull request #1923 from GPCsolutions/fixws

Fixed invoices webservice

eldy committed:
Merge pull request #1925 from tiaris/develop

adding hooks for supplier card and order card

FHenry committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into
develop

Conflicts:
	htdocs/contrat/class/contrat.class.php
	htdocs/fourn/card.php

eldy committed:
Bad place of code

FHenry committed:
Fix travis (strange I didn't update this file)

rdoursenaud committed:
Allow reporting progress along with timespent

eldy committed:
Merge branch 'develop-accounting' of github.com:aspangaro/dolibarr into
develop

Conflicts:
	htdocs/compta/ventilation/card.php
	htdocs/compta/ventilation/fournisseur/card.php
	htdocs/compta/ventilation/fournisseur/list.php
	htdocs/compta/ventilation/list.php
	htdocs/core/menus/init_menu_auguria.sql

eldy committed:
Merge pull request #1800 from Devensys/Develop-HighlitingIBAN

Highliting IBAN (sepa)

hregis committed:
Fix: uniformize code

eldy committed:
Fix: checkstyle

hregis committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

Xebax committed:
Fix bug #1604: When a member is created, the birthday is not recorded.

rdoursenaud committed:
Suggest dependencies needed by PHPExcel

Xebax committed:
Fix bug #1610: Civility is not displayed.

hregis committed:
FIXME $key is not defined (strict mode)

hregis committed:
FIXME $socid is not defined (strict mode)

hregis committed:
Fix: missing fields

hregis committed:
Fix: strict mode paradise

eldy committed:
Uniformize, support, install and login page with a more neutral
background (nearest bootstrap). Remove image background.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Solve fixme.

eldy committed:
Fix: Avoid error ErrorBadValueForParamNotAString

eldy committed:
Merge pull request #1934 from GPCsolutions/deps

Suggest dependencies needed by PHPExcel

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge pull request #1932 from Xebax/develop

Fix bug #1604: "When a member is created, the birthday is not recorded" and bug #1610: "Civility is not displayed".

eldy committed:
Fix: Translation

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Brighter button. We use CSS and no more image.

eldy committed:
Fix: miscellaneous bugs

fappels committed:
Expedition lines from same orderline merged.

When there are more expedition lines from same orderline key, merge them
and get source warehouses. Usefull future possibility for creating
shippinglines with same products from multiple warehouses.
@KreizIT also changed some batch related items.

fappels committed:
Fix deliveryorder qty and remainingqty positions

eldy committed:
Add missing picto

eldy committed:
Minor look enhancement

fappels committed:
Fix migration sql for inserting accounting records

Migration SQL could not insert accounting records because datec was set
to NOT NULL

eldy committed:
Add hook formatEvent and fix bug into per user view.

eldy committed:
Fix: edit event
New: Can use color of type of event into dictionnary.

eldy committed:
Fix: Event update

eldy committed:
Sync from transifex

eldy committed:
Merge pull request #1916 from FHenry/develop

Add feature to order to invoice on supplier part

eldy committed:
Merge pull request #1937 from fappels/develop

Expedition lines from same orderline merged.

eldy committed:
Sync from transifex.
Fix var not initialized.

eldy committed:
Uniformize code: property "nom" for thirdparty is aliased into "name"
everywhere.

eldy committed:
New: Can search list of thirdparties from web service on part of name.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into
develop

Conflicts:
	htdocs/contrat/list.php

eldy committed:
Fix scrutinizer

eldy committed:
Fix: Pb when showing availability

eldy committed:
Fix: Restore broken feature. One hook = One call of function per module.

eldy committed:
Fix: Edition of event

eldy committed:
Fix translation

eldy committed:
Fix translation

eldy committed:
Fix trad and sql error

eldy committed:
Change comented log to help debug

eldy committed:
Fix: miss dol_fiche_end

hregis committed:
New: add context "globalcard" for declaring only once all cards

hregis committed:
Fix: missing $ismultientitymanaged

fappels committed:
Fix increase stock on shipment validation

$obj not set.
Re-enabled delete button for shipment with batches, was increase issue,
not decrease issue.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

Commit:
Fix: Showing accounting account fails

aspangaro committed:
Fix :: Button on/off for account in chart of accounts

aspangaro committed:
Merge remote-tracking branch 'Upstream/develop' into develop-accounting

eldy committed:
Fix: Warning sur traduction

eldy committed:
New: Can add product into category from category card.

aspangaro committed:
Presentation

hregis committed:
Fix: add constant MAIN_OVERWRITE_THEME_PATH if the theme does not have
the same name as the module

eldy committed:
Fix: button must be align to right

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
More comment to explain and remove a todo.

eldy committed:
Fix: Avoid duplicate error message
Fix: pgsql syntax error

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

aspangaro committed:
Merge remote-tracking branch 'Upstream/develop' into develop-accounting

eldy committed:
Style for install match css of login.

eldy committed:
Some fix on per user view.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Sync transifex

eldy committed:
Fix: CRLF

eldy committed:
Fix: bad parameter

eldy committed:
Fix: Bad length of field
Transifex sync

defrance committed:
Update product.class.php

Label of product on tooltip is more usefull than the ref who's already on the link

aspangaro committed:
Merge remote-tracking branch 'Upstream/develop' into develop-accounting

Commit:
Add function hideMail and use

Commit:
copyright and annotation

Commit:
Add support tld with dots

hregis committed:
Fix: avoid warning

Commit:
Trad: Syncro es_ES from transifex

Commit:
Trad: Syncro es_ES/compta.lang from transifex

jfefe committed:
Allow to make PUT request with getURLContent() function

eldy committed:
Sync transifex

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into
develop

Conflicts:
	htdocs/langs/es_ES/accountancy.lang
	htdocs/langs/es_ES/admin.lang
	htdocs/langs/es_ES/agenda.lang
	htdocs/langs/es_ES/categories.lang
	htdocs/langs/es_ES/companies.lang
	htdocs/langs/es_ES/compta.lang
	htdocs/langs/es_ES/donations.lang
	htdocs/langs/es_ES/errors.lang
	htdocs/langs/es_ES/holiday.lang
	htdocs/langs/es_ES/interventions.lang
	htdocs/langs/es_ES/projects.lang
	htdocs/langs/es_ES/sendings.lang
	htdocs/langs/es_ES/suppliers.lang

eldy committed:
Clean deprecated code.

eldy committed:
Fix: translation for vi_VN

eldy committed:
Fix: File names must be in lowercase. 
Fix: Missing includes.

eldy committed:
Merge pull request #1953 from jfefe/patch-6

Allow to make PUT request with getURLContent() function

eldy committed:
Merge branch 'develop-HiddenMailPasswordForgotten' of
github.com:Devensys/dolibarr into
Devensys-develop-HiddenMailPasswordForgotten

Conflicts:
	htdocs/core/lib/functions.lib.php

eldy committed:
Merge

eldy committed:
git push origin developMerge branch 'Devensys-develop-HiddenMailPasswordForgotten' into develop

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge pull request #1930 from GPCsolutions/reportprogress

Allow reporting progress along with timespent

marcosgdf committed:
Lowecased booleans to fit with PSR-2 specification.

eldy committed:
Qual: Uniformize translated label of progress on all pages.
Fix: Color of first line of documents.
Fix: The calculated progress was not calculated.

eldy committed:
Prepare code to manage task_datehour with task_date.

eldy committed:
Merge pull request #1954 from marcosgdf/lowecased-booleans

Lowercased booleans to fit with PSR-2 specification.

eldy committed:
Merge pull request #1949 from defrance69/patch-23

Update product.class.php

eldy committed:
Merge pull request #1942 from fappels/develop

Fix increase stock on shipment validation

eldy committed:
Merge pull request #1944 from aspangaro/develop-accounting

Fix :: Button on/off for account in chart of accounts

eldy committed:
Merge pull request #1861 from defrance69/patch-20

Update modContrat.class.php

eldy committed:
Some fix with export definition. Works now with pgql too.

eldy committed:
Merge remote-tracking branch 'origin/3.6' into develop

Conflicts:
	ChangeLog
	htdocs/categories/fiche.php
	htdocs/comm/action/index.php
	htdocs/comm/propal.php
	htdocs/commande/fiche.php
	htdocs/compta/facture.php
	htdocs/compta/facture/class/facture.class.php
	htdocs/contrat/class/contrat.class.php
	htdocs/core/class/commonobject.class.php
	htdocs/core/tpl/objectline_create.tpl.php
	htdocs/fourn/fiche.php
	htdocs/product/composition/fiche.php
	htdocs/product/price.php
	htdocs/projet/tasks/task.php
	htdocs/projet/tasks/time.php

eldy committed:
Fix: area to generate doc must use half size not onethird.
Fix: Missing maxwidthonsmartphone.
Fix: Missing backtopage
Fix: Missing translation key

eldy committed:
New: Add number of active notification into tab title (like we do for
notes and documents)

eldy committed:
Better contrast

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

marcosgdf commented:
This PR is broken. It is merging develop into 3.6.

eldy committed:
Qual: Removed a lot of duplicate code
New: extrafields for projects and tasks are exported to ODT documents.

eldy committed:
Add index on new table llx_c_email_template

eldy committed:
Use language to find template email.

eldy committed:
Qual: Indentation muse be done with css and not using multiple &nbsp;
spaces.

eldy commented:
PR says "merge into 3.6 from develop".
This a probably a bad manipulation, we never merge code from develop into a lower version.
